rules/terraform_module_shallow_clone_test.go

```diff
@@ -0,0 +1,170 @@
+package rules
+
+import (
+	"testing"
+
+	hcl "github.com/hashicorp/hcl/v2"
+	"github.com/terraform-linters/tflint-plugin-sdk/helper"
+)
+
+func Test_TerraformModuleShallowClone(t *testing.T) {
+	cases := []struct {
+		Name     string
+		Content  string
+		Expected helper.Issues
+	}{
+		{
+			Name: "terraform registry module",
+			Content: `
+module "registry" {
+  source = "hashicorp/consul"
+  version = "~> 1.0"
+}`,
+			Expected: helper.Issues{},
+		},
+		{
+			Name: "local module",
+			Content: `
+module "local" {
+  source = "./modules/consul"
+}`,
+			Expected: helper.Issues{},
+		},
+		{
+			Name: "unpinned git module",
+			Content: `
+module "unpinned" {
+  source = "git://github.com/hashicorp/consul.git"
+}`,
+			Expected: helper.Issues{},
+		},
+		{
+			Name: "git module with raw commit",
+			Content: `
+module "commit_pinned" {
+  source = "git://github.com/hashicorp/consul.git?ref=abc123def456789012345678901234567890abcd"
+}`,
+			Expected: helper.Issues{},
+		},
+		{
+			Name: "git module with shallow clone enabled (ref first)",
+			Content: `
+module "shallow_clone" {
+  source = "git://github.com/hashicorp/consul.git?ref=v1.0.0&depth=1"
+}`,
+			Expected: helper.Issues{},
+		},
+		{
+			Name: "git module with shallow clone enabled (depth first)",
+			Content: `
+module "shallow_clone" {
+  source = "git://github.com/hashicorp/consul.git?depth=1&ref=v1.0.0"
+}`,
+			Expected: helper.Issues{},
+		},
+		{
+			Name: "pinned git module with SSH protocol",
+			Content: `
+module "ssh_pinned" {
+  source = "git::ssh://git@github.com/hashicorp/consul.git?ref=v1.0.0"
+}`,
+			Expected: helper.Issues{
+				{
+					Rule:    NewTerraformModuleShallowCloneRule(),
+					Message: "Module source \"git::ssh://git@github.com/hashicorp/consul.git?ref=v1.0.0\" should enable shallow cloning by adding \"depth=1\" parameter",
```

Comment:
Use backticks to minimize the need for escaping in strings with embedded double quotes